src/components/NavBar.tsx

```diff
@@ -9,7 +9,12 @@ function NavBar() {
   const pageName = url.substring(url.lastIndexOf('/') + 1)
```

Comment:
Not under your PR right now but noticed this. This feels very hacky and could break if there are arguments in the url like `?arg1=helloworld` or fragments which helps user scroll like `/route#chapter2` etc. Not sure what the correct way to do this would be but please look into it. I'll try to find too.

Reply:
One suggestion I can give is to somehow check against the `PATHS` enum under `PATHS.ts` in the `routes` folder. 

Perhaps we can make a helper function to help check the strings properly and not include the arguments in the URL, with the use of a `.contains()` method.  Not sure if I'm missing out any edge cases though.

@tinenhao what do you think?

Reply:
fixed